lib/nerves_hub_link/socket.ex

```diff
@@ -296,6 +298,27 @@ defmodule NervesHubLink.Socket do
     {:ok, set_iex_timer(socket)}
   end
 
+  def handle_message(@console_topic, "file-data/start", params, socket) do
+    :ok = File.mkdir_p!(socket.assigns.upload_path)
+    path = Path.join(socket.assigns.upload_path, params["filename"])
+    _ = File.rm_rf!(path)
+    :ok = File.touch!(path)
+    {:ok, socket}
+  end
+
+  def handle_message(@console_topic, "file-data", params, socket) do
+    path = Path.join(socket.assigns.upload_path, params["filename"])
+    file = File.open!(path, [:append])
+    chunk = Base.decode64!(params["data"])
+    :ok = IO.binwrite(file, chunk)
+    :ok = File.close(file)
```

Comment:
```suggestion
    file = File.open!(path, [:append], fn fd ->
        chunk = Base.decode64!(params["data"])
        IO.binwrite(fd, chunk)
    end)
```

It's prob worth using the function to `File.open!/3` here so that if there is an error it will definitely get closed